src/firefly/js/visualize/ui/ApiExpandedDisplay.jsx

```diff
@@ -33,8 +34,10 @@ export class ApiExpandedDisplay extends Component {
     }
 
     storeUpdate() {
-        if (visRoot()!==this.state.visRoot || lastMouseCtx() !==this.state.currMouseState) {
-            this.setState({visRoot:visRoot(), currMouseState:lastMouseCtx()});
+        if (visRoot()!==this.state.visRoot || 
+            lastMouseCtx() !==this.state.currMouseState || 
+            readoutRoot()!==this.state.readout) {
+            this.setState({visRoot:visRoot(), currMouseState:lastMouseCtx(), readout:readoutRoot()});
```

Comment:
Ultimately I don't think it matters for performance. Maybe it would if it were called in a loop.  Even if you think about performance, most of the time the if would be `false`.  Therefore putting it is 3 variables might take more resources.